Content.Shared/AME/SharedAmeControllerComponent.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Robust.Shared.Serialization;
+
+namespace Content.Shared.AME;
+
+[Virtual]
+public class SharedAmeControllerComponent : Component
+{
+    [Serializable, NetSerializable]
```

Comment:
i dont think bui state being inside the component is good, might be wrong

Reply:
Looks like putting the state definition in the shared component is one of a few conflicting practices. I've seen it a couple other places.

Reply:
Yeah, just make it share the namespace, dont put it inside there